GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -2698,6 +2701,12 @@ private void editFileToolStripMenuItem_Click(object sender, EventArgs e)
 
         private void CommitAndPush_Click(object sender, EventArgs e)
         {
+            if (CommitAndPush.Text == _push.Text)
```

Comment:
Slightly suspicious to compare texts, but it is the easiest way to achieve this...

Reply:
That's what I though, too.